Sources/VaporJWT/Payload.swift

```diff
@@ -0,0 +1,8 @@
+import Node
+import Auth
+
+/// Types conforming to this protocol
+/// can be stored as payloads in JWTs
+public protocol Payload: NodeConvertible {
+    func makeCredentials() throws -> Auth.Credentials
```

Comment:
Would this be a good place to add `var claims: [Claim]`? The default implementation would return `[]`.

In `LoginMiddleware` we could then add something like:
```
token.verifyClaims(payload.claims)
```
to support verifying claims as well as signatures.